jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.api;
+// TODO: Move to com.google.cloud.tools.jib
+
+import com.google.cloud.tools.jib.configuration.LayerConfiguration;
+import com.google.cloud.tools.jib.configuration.Port;
+import com.google.cloud.tools.jib.image.ImageReference;
+import com.google.common.collect.ImmutableList;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Builds a container with Jib.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * Jib.from(baseImage)
+ *    .addFiles(sourceFiles, extractionPath)
+ *    .setEntrypoint(entrypoint)
+ *    .setProgramArguments(arguments)
+ *    .addEnvironmentVariable(key, value)
+ *    .addExposedPort(
+ *    .addLabel(key, value)
+ *    .containerize(...);
+ * }</pre>
+ */
+public class JibContainerBuilder {
+
+  private final ImageReference baseImageReference;
+
+  private List<LayerConfiguration> layerConfigurations = new ArrayList<>();
+  private Map<String, String> environment = new HashMap<>();
+  private List<Port> ports = new ArrayList<>();
+  private Map<String, String> labels = new HashMap<>();
+  @Nullable private ImmutableList<String> entrypoint;
+  @Nullable private ImmutableList<String> programArguments;
+
+  /** Instantiate with {@link Jib#from}. */
+  JibContainerBuilder(ImageReference baseImageReference) {
```

Comment:
Should this be private or `@VisibleForTesting` if `Jib#from` is supposed to be used instead?

Reply:
This is package-private for `Jib#from` to call it.